internal/credential/verification.go

```diff
@@ -5,86 +5,145 @@ import (
 // In the future the set of verification checks will be configurable.
-func NewCredentialVerifier(resolver *didsdk.Resolver) (*CredentialVerifier, error) {
-	if resolver == nil {
-		return nil, errors.New("resolver is nil")
+func NewCredentialVerifier(didResolver *didsdk.Resolver, schemaResolver schema.Resolution) (*Verifier, error) {
+	if didResolver == nil {
+		return nil, errors.New("didResolver cannot be nil")
+	}
+	if schemaResolver == nil {
+		return nil, errors.New("schemaResolver cannot be nil")
 	}
 	// TODO(gabe): consider making this configurable
 	verifiers := verification.KnownVerifiers
 	verifier, err := verification.NewCredentialVerifier(verifiers)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create static credential verifier")
 	}
-	return &CredentialVerifier{
-		verifier: verifier,
-		resolver: resolver,
+	return &Verifier{
+		verifier:       verifier,
+		didResolver:    didResolver,
+		schemaResolver: schemaResolver,
 	}, nil
 }
 
 // TODO(gabe) consider moving this verification logic to the sdk https://github.com/TBD54566975/ssi-service/issues/122
 
 // VerifyJWTCredential first parses and checks the signature on the given JWT credential. Next, it runs
 // a set of static verification checks on the credential as per the credential service's configuration.
-func (v CredentialVerifier) VerifyJWTCredential(token string) error {
+func (v Verifier) VerifyJWTCredential(token string) error {
+	// first, parse the token to see if it contains a valid verifiable credential
 	cred, err := signing.ParseVerifiableCredentialFromJWT(token)
 	if err != nil {
 		return util.LoggingErrorMsg(err, "could not parse credential from JWT")
 	}
+
+	// TODO(gabe): perhaps this should be a verification method referenced on the proof object, not the issuer
 	issuerDID := cred.Issuer.(string)
-	resolved, err := v.resolver.Resolve(issuerDID)
+
+	// next, get the verification information (key) from the did document of the issuer
+	resolved, err := v.didResolver.Resolve(issuerDID, nil)
 	if err != nil {
 		return errors.Wrapf(err, "failed to resolve did: %s", issuerDID)
 	}
+
+	// next, get the verification information (key) from the did document of the issuer
 	kid, pubKey, err := keyaccess.GetVerificationInformation(resolved.DIDDocument, "")
 	if err != nil {
 		return util.LoggingErrorMsg(err, "could not get verification from credential")
 	}
+
+	// construct a signature verifier from the verification information
 	verifier, err := keyaccess.NewJWKKeyAccessVerifier(kid, pubKey)
 	if err != nil {
 		return util.LoggingErrorMsg(err, "could not create verifier")
 	}
+
+	// verify the signature on the credential
 	if err := verifier.Verify(keyaccess.JWKToken{Token: token}); err != nil {
 		return util.LoggingErrorMsg(err, "could not verify credential's signature")
 	}
-	if err = v.verifier.VerifyCredential(*cred); err != nil {
+
+	// if the credential has a schema, resolve it before it is to be used in verification
```

Comment:
same code block on line 131, you could make into a single function

Reply:
good idea